File: id/src/main/java/com/simprints/id/activities/front/FrontPresenter.java
package com.simprints.id.activities.front;


import android.support.annotation.NonNull;

class FrontPresenter implements FrontContract.Presenter {

    @NonNull
    private final FrontContract.View frontView;

    /**
     * @param view The FrontActivity
     */
    FrontPresenter(@NonNull FrontContract.View view) {
        frontView = view;
        frontView.setPresenter(this);
    }

    @Override
    public void start() {
    }
}

File: id/src/main/java/com/simprints/id/activities/front/FrontPresenter.kt
package com.simprints.id.activities.front

import com.simprints.id.data.secure.SecureDataManager

internal class FrontPresenter(val view: FrontContract.View, override val secureManager: SecureDataManager) : FrontContract.Presenter {

    init {
        view.setPresenter(this)
    }

    override fun start() {
    }

    override fun doSecurityChecks(): Boolean {
        if (secureManager.getProjectSecretOrEmpty() == "") {
            view.openRequestAPIActivity()
            return false
        }
        return false
    }
}
